app/scripts/modules/core/src/managed/ArtifactsList.tsx

```diff
@@ -87,3 +80,85 @@ export const ArtifactRow = ({
     </div>
   );
 };
+
+const MAX_STATUS_BUBBLE_COUNT = 3;
+
+const ArtifactStatuses = ({
+  isHovered,
+  isSelected,
+  version,
+}: {
+  isHovered: boolean;
+  isSelected: boolean;
+  version: IManagedArtifactVersion;
+}) => {
+  const statuses = getArtifactStatuses(version);
+
+  // We only have limited space in the UI to show the bubbles related to the artifact, so only take
+  // the first few important statuses.
+  const statusBubblesToRender = take(statuses, MAX_STATUS_BUBBLE_COUNT - 1);
+  if (statusBubblesToRender.length < statuses.length) {
+    statuses.length - statusBubblesToRender.length > 1
+      ? // If there's more than 1 hidden status bubble, then show the count bubble.
+        statusBubblesToRender.push({ kind: 'count', count: statuses.length - statusBubblesToRender.length })
+      : // If there's only one hidden status bubble, then don't hide it.
+        statusBubblesToRender.push(statuses[statuses.length - 1]);
+  }
+  return (
+    <div className="flex-container-h middle sp-margin-s-right">
+      {statusBubblesToRender.map((status, index) => {
+        if (status.kind === 'status') {
+          return (
+            <div
+              className={classNames(styles.ArtifactStatus, {
+                [styles.selected]: isSelected,
+                [styles.hovered]: isHovered,
+              })}
+              key={status.iconName}
```

Comment:
The way it is currently implemented, there will be only one status bubble 'cos we're running through all environments to see if a condition is satisfied in any of the environment. This could be modified to add the count at the bottom right of the bubble, but that might affect the layout a little bit with all the stacking. Would definitely need @gcomstock's input on this one :)